executor/predictor/predictor_process_test.go

```diff
@@ -455,3 +463,85 @@ func TestModelWithLogResponses(t *testing.T) {
 	g.Expect(smRes.GetData().GetNdarray().Values[1].GetNumberValue()).Should(Equal(2.0))
 	g.Eventually(func() bool { return logged }).Should(Equal(true))
 }
+
+func TestModelWithLogRequestsNilPUIDError(t *testing.T) {
+	t.Logf("Started")
+	g := NewGomegaWithT(t)
+	modelName := "foo"
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		//g.Expect(r.Header.Get(logger.CloudEventsIdHeader)).Should(Equal(testEventId))
+		g.Expect(r.Header.Get(logger.CloudEventsTypeHeader)).To(Equal(logger.CEInferenceRequest))
+		g.Expect(r.Header.Get(logger.CloudEventsTypeSource)).To(Equal(testSourceUrl))
+		g.Expect(r.Header.Get(modelIdHeaderName)).To(Equal(modelName))
+		g.Expect(r.Header.Get(contentTypeHeaderName)).To(Equal(grpc.ProtobufContentType))
+		g.Expect(r.Header.Get(requestIdHeaderName)).To(Equal(testSeldonPuid))
+		w.Write([]byte(""))
+		fmt.Printf("%+v\n", r.Header)
+		fmt.Printf("%+v\n", r.Body)
+	})
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	logf.SetLogger(logf.ZapLogger(false))
+	log := logf.Log.WithName("entrypoint")
+	logger.StartDispatcher(1, log, "", "", "")
+
+	model := v1.MODEL
+	graph := &v1.PredictiveUnit{
+		Name: modelName,
+		Type: &model,
+		Endpoint: &v1.Endpoint{
+			ServiceHost: "foo",
+			ServicePort: 9000,
+			Type:        v1.REST,
+		},
+		Logger: &v1.Logger{
+			Mode: v1.LogRequest,
+			Url:  &server.URL,
+		},
+	}
+
+	_, err := createPredictorProcessWithoutPUID(t).Predict(graph, createPredictPayload(g))
+	g.Expect(err).NotTo(BeNil())
+	g.Expect(err.Error()).Should(Equal("context value Seldon PUID Header is nil: interface to string conversion failed"))
+}
+
+func TestModelWithLogResponsesNilPUIDError(t *testing.T) {
```

Comment:
seems to be almost a duplicate of above test - can't they be combined?

Reply:
Fixed